src/platform/packages/shared/response-ops/rule_form/src/rule_definition/rule_dashboards.tsx

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React, { useState, useEffect, useCallback } from 'react';
+import { EuiComboBox, EuiSplitPanel, EuiFlexItem, EuiSpacer, EuiTitle } from '@elastic/eui';
+import { RuleTypeParams } from '@kbn/alerting-types';
+import { RuleFormPlugins } from '../types';
+import { dashboardServiceProvider, type DashboardItem } from './dashboard_service';
+import { useRuleFormState, useRuleFormDispatch } from '../hooks';
+import { ALERT_LINK_DASHBOARDS_TITLE } from '../translations';
+
+interface RuleFormPluginsProps {
+  plugins: Pick<RuleFormPlugins, 'dashboard' | 'featureFlags'>;
+}
+
+interface DashboardOption {
+  value: string;
+  label: string;
+}
+
+interface RuleTypeParamsWithDashboards extends RuleTypeParams {
+  dashboards?: Array<{ id: string }>;
+}
+
+export const RuleDashboards = ({ plugins }: RuleFormPluginsProps) => {
+  const { featureFlags, dashboard: dashboardService } = plugins;
+  const { formData } = useRuleFormState();
+  const dispatch = useRuleFormDispatch();
+  const params = formData.params as RuleTypeParamsWithDashboards;
```

Comment:
@benakansara While cleaning up my PR, I realized I save linkedDashboards under params. Let's catch on this, because in your initial POC you followed another approach